### PR TITLE
Allow to use network port with VLAN-subinterfaces for untagged traffic

### PR DESCRIPTION
**Pull request by milan-zededa**
With this enhancement, user will be able to use VLAN parent interface as an L3 endpoint and access the untagged network either for EVE management or from an application via Local network instance.

Some changes were required to allow having SystemAdapter assigned to a network port (physical or bond) which is at the same time referenced by a VLAN sub-interface. Previously, this was not allowed and resulted in parsing errors published to the controller.

In terms of the Linux network configuration, not many changes were required. We simply allow assigning static IP or running DHCP client even for VLAN parent interface.
The only scenario which is not allowed, is to bridge network port and at the same time use it with VLAN sub-interfaces. This means that accessing untagged network from a Switch NI is not allowed when the port is already used with VLAN subinterfaces.

Submitted example to eden: https://github.com/lf-edge/eden/pull/1043 (later will be rewritten to Go test)

### CODE REVIEW

File: pkg/pillar/dpcreconciler/linux.go [hunk 1]
```diff
@@ -965,11 +965,15 @@ func (r *LinuxDpcReconciler) getIntendedPhysicalIfs(dpc types.DevicePortConfig)
 			}
 		case types.L2LinkTypeVLAN:
 			parent := dpc.LookupPortByLogicallabel(port.VLAN.ParentPort)
+			usage := generic.IOUsageVlanParent
+			if parent.IsL3Port {
+				usage = generic.IOUsageVlanParentAndL3Adapter
+			}
```
Comment (eriknordmark) [inline pkg/pillar/dpcreconciler/linux.go:971]: Does this imply that you have slightly redefined what IsL3Port means? 
Or am I not understanding the IOUsage* - Does IOUsageX mean that it is currently used for X, or does it mean that it is available to be used for X?

For example, does IOUsageVlanParentAndL3Adapter mean that it is used as an L3Adapter or that it is capable/available to be used as an L3Adapter?

File: pkg/pillar/dpcreconciler/linuxitems/bond.go [hunk 2]
```diff
@@ -320,7 +323,10 @@ func (c *BondConfigurator) Delete(ctx context.Context, item depgraph.Item) error
 func (c *BondConfigurator) NeedsRecreate(oldItem, newItem depgraph.Item) (recreate bool) {
 	oldBondCfg := oldItem.(Bond)
 	newBondCfg := newItem.(Bond)
-	if !reflect.DeepEqual(oldBondCfg.BondConfig, newBondCfg.BondConfig) {
+	if oldBondCfg.LacpRate != newBondCfg.LacpRate ||
```
Comment (eriknordmark) [inline pkg/pillar/dpcreconciler/linuxitems/bond.go:326]: Is this a separate bug fix or related to the VLAN change?

**Review by eriknordmark — APPROVED** [review]
Some questions about naming/comments.
Kicking off tests in parallel.

FWIW I ran it through my ztests setup - primarily to check if the old hardware models which don't have logicallabel set would run into trouble. No issues found.

File: pkg/pillar/dpcreconciler/linux.go (hunk 1 above)
Comment (milan-zededa) [reply pkg/pillar/dpcreconciler/linux.go:971]: Both `IsL3Port` and the `IOUsage` from DPCReconciler describe the current usage of the adapter (not what it is available for but what it is actually used for in the current DPC).

Comment (milan-zededa) [reply pkg/pillar/dpcreconciler/linux.go:971]: In the case of the code referenced, we see that the port is used with SystemAdapter (i.e. has IP config) and at the same time has VLAN sub-interface, so usage is `IOUsageVlanParentAndL3Adapter` (but note that this IOUsage is an internal type inside DPCReconciler, not part of some pubsub message that would be visible to a user).

File: pkg/pillar/dpcreconciler/linuxitems/vlan.go [hunk 3]
```diff
@@ -28,6 +28,9 @@ type Vlan struct {
 	ParentIfName string
 	// ParentL2Type : link type of the parent interface (bond or physical).
 	ParentL2Type types.L2LinkType
+	// ParentIsL3Port is true when the parent port is used both as a VLAN parent
```
[the remark this answers is not in the document]
Comment (milan-zededa) [reply pkg/pillar/dpcreconciler/linuxitems/vlan.go:31]: Used

File: pkg/pillar/dpcreconciler/linuxitems/bond.go (hunk 2 above)
Comment (milan-zededa) [reply pkg/pillar/dpcreconciler/linuxitems/bond.go:326]: This is just to optimize comparison and avoid using reflect.
I'm going to put this into a separate commit...

Comment (milan-zededa) [reply pkg/pillar/dpcreconciler/linuxitems/bond.go:326]: Done

**Pull request merged**